File: diagonal_negativos.py
#Fazer um programa para ler um número inteiro N (máximo = 10) e uma matriz quadrada de ordem N
#contendo números inteiros. Em seguida, mostrar a diagonal principal e a quantidade de valores
#negativos da matriz.

n = int(input("Qual a ordem da matriz?"))
mat = [[0 for x in range(n)] for x in range(n)]

for i in range (0, n):
    for j in range (0, n):
        mat[i][j] = int(input(f"Elemento[{i},{j}]: "))

print("Diagonal principal: ")
for i in range (0, n):
    print(f"{mat[i][i]} ", end ="")

cont = 0
for i in range (0, n):
    for j in range (0, n):
        if mat[i][j] < 0:
            cont = cont + 1

print()
print(f"Quantidade de negativos = {cont}")

File: workSpace Java/curso_programaçao/src/diagonal_negativos.java
import java.util.Scanner;
public class diagonal_negativos {
	//Fazer um programa para ler um número inteiro N (máximo = 10) e uma matriz quadrada de ordem N
	//contendo números inteiros. Em seguida, mostrar a diagonal principal e a quantidade de valores
	//negativos da matriz.
	public static void main(String[] args) {
		
		Scanner sc = new Scanner(System.in);
		
		int n, cont;
		
		System.out.print("Qual a ordem da matriz? ");
		n = sc.nextInt();
		
		int[][] mat = new int[n][n];
		
		for (int i = 0; i < n; i++) {
			for (int j = 0; j < n; j++) {
				System.out.print("Elemento [" + i +" " + j + "]: " );
				mat[i][j] = sc.nextInt();
			}
		}
		
		System.out.println();
		System.out.println("Diagonal principal: ");
		for (int i = 0; i < n; i++) {
			System.out.print( + mat[i][i] + " ");
		}
		
		cont = 0;
		System.out.println();
		System.out.print("Quantidade de negativos = ");
		for (int i = 0; i < n; i++) {
			for (int j = 0; j < n; j++) {
				if (mat[i][j] < 0) {
					cont++;
				}
			}
		}
	   
		System.out.println(+ cont);
		sc.close();
	
	}
}
